docs/proposals/multiple-sources-for-applications-ui.md

```diff
@@ -0,0 +1,225 @@
+and put into a new tab called **Sources**, and it will be combined with the **Parameters** tab (more details following). 
+The new **Sources** tab will allow users to view all the information related to each source, including the repo URL 
+and path, chart and revision for Helm, etc. 
+
+The view should show one source at a time (similar to what the UI is doing now, which only shows one source), but with 
+widgets to allow users to cycle (via pagination or combo selector?) through each source. There are API calls to retrieve 
+the data for each source.
+
+<img height="50%" width="50%" src="images/new-sources-tab.png"/>
+
+Figure 2. The new SOURCES tab will allow access to view all sources and application parameters.
+
+#### ii) Parameters Tab
+_Current Behavior:_ 
+
+The Parameters tab shows the application parameters for the application’s repository details type or source. These can 
+be Helm, Kustomize, Directory or Plugin (CMP).
+
+_Proposed Change:_
+
+The Parameter tab will be removed but the contents of the current parameters tab will be ‘reused’ and will be shown in 
+the new **SOURCES** tab as described above. The parameters and parameter values will be shown for whatever source is 
+selected by the user.
+
+#### iii) Update/Edit Capability in the New Sources Tab
+
+The above points describe how all the sources will be rendered.  However, the Sources tab should be the page to allow 
+users to delete and add sources. (You can currently change the repo URL and path from the Summary tab, or manually edit 
+the application by hand, in the Manifest tab, but this is not considered as ‘guided’ editing).
+
+_Current Behavior:_ 
+
+The current form-based UI doesn’t support deleting a chosen/desired source of a multi-source application. It certainly 
+does not obviously support deleting the only source there is in a single-source application.
+
+_Proposed Change:_
+
+In addition to adding the new SOURCES tab from section i) and ii), two new buttons (_Add Source_ and _Delete Source_) will 
+be added to the page. For the _Add Source_ button, a separate dialog/panel will need to appear to allow the user to
+input the parameters or other information.
+
+Validation of any newly added source should prevent users from adding the same resource, and prevent users from
+deleting all sources, etc.
+
+### History and Rollback
+
+Current Behavior: The History and Rollback button for multi-source apps is disabled. It's only enabled
+for single-source apps, and shows source information as shown in Figure 3.
+
+<img height="50%" width="50%" src="images/history-rollback-contents.png"/>
+
+Figure 3: Source information in History
+
+Jorge has submitted a PR for rollback which includes controller and UI changes [here](https://github.com/argoproj/argo-cd/pull/14124). 
+This can be treated as a separate, independent proposal.
+
+Other related changes pertain to the Last Synced Details. The Sync Details panel needs to be updated to show sync info
```

Comment:
Info in `SYNC STATUS`  will also display multiple sources?

Reply:
@ashutosh16 , the SYNC STATUS header content might not have to change, but the "more information" panel probably will have to show the sources information, just like what is presented in the rollback and history.